api/pkg/db/initializer/initializer.go

```diff
@@ -196,3 +204,70 @@ func withTransaction(db *gorm.DB, log *log.Logger, data *app.Data, fns ...initFn
 	txn.Commit()
 	return nil
 }
+
+func addApiServerUser(db *gorm.DB, log *log.Logger) error {
+
+	newUser := model.User{
+		Type:  "agent",
+		Email: apiserverEmailID,
+	}
+
+	newUser.ID = uint(apiserverID)
+	if err := db.Create(&newUser).Error; err != nil {
+		log.Error(err)
+		return err
+	}
+
+	apiAccount := model.Account{
+		UserID:    uint(apiserverID),
+		UserName:  apiserverUserName,
+		Name:      apiserverName,
+		AvatarURL: "",
+		Provider:  "github",
+	}
+
+	if err := db.Create(&apiAccount).Error; err != nil {
+		log.Error(err)
+		return err
+	}
+
+	return nil
+}
+
+// AddResources func creates an apiserver account and adds resources from catalog to database
+func (i *Initializer) AddResources(db *gorm.DB, api app.BaseConfig, logger *log.Logger) error {
+
+	// Check if apiserver account exists or not
+	// If not it creates apiserver account with catalog refresh scope
+	q := db.Model(&model.Account{}).Where("user_name = ?", apiserverUserName)
+	apiAccount := model.Account{}
+
+	if err := q.First(&apiAccount).Error; err == gorm.ErrRecordNotFound {
+
+		logger.Infof("user %s account not found,creating account for apiserver", "apiserver")
+
+		if err := addApiServerUser(db, logger); err != nil {
+			logger.Error(err)
+			return err
+		}
+	}
```

Comment:
I think `AddResources` function does lot more than it is supposed to do, so we can move this logic into a separate function as it is checking if the user exists, if not it adds an `apiServer` user

In short we can add a function something like this 

```golang
func checkIfApiServerAccountExists(db *gorm.DB, logger *log.Logger) error {
	// Check if apiserver account exists or not
	// If not, it creates apiserver account with catalog refresh scope
	q := db.Model(&model.Account{}).Where("user_name = ?", apiserverUserName)
	apiAccount := model.Account{}

	if err := q.First(&apiAccount).Error; err == gorm.ErrRecordNotFound {

		logger.Infof("user %s account not found,creating account for apiserver", "apiserver")

		if err := addApiServerUser(db, logger); err != nil {
			logger.Error(err)
			return err
		}
	}

	return nil
}

func (i *Initializer) AddResources(db *gorm.DB, api app.BaseConfig, logger *log.Logger) error {

	if err := checkIfApiServerAccountExists(db, logger); err != nil {
		return err
	}
        
        ... 
        other part of the logic....
        ......
}

```